### PR TITLE
Fix compliance logic for off-ledger certification

### PR DESCRIPTION
**Pull request by spivachuk**
- Corrected compliance logic for off-ledger certification approach.
- Updated tests and documentation correspondingly.

### CODE REVIEW

File: x/compliance/keeper/msg_server_revoke_model.go [hunk 1]
```diff
@@ -28,6 +29,17 @@ func (k msgServer) RevokeModel(goCtx context.Context, msg *types.MsgRevokeModel)
...
+	modelVersion, found := k.modelKeeper.GetModelVersion(ctx, msg.Vid, msg.Pid, msg.SoftwareVersion)
+	if !found {
+		return nil, modeltypes.NewErrModelVersionDoesNotExist(msg.Vid, msg.Pid, msg.SoftwareVersion)
+	}
+
+	// check if softwareVersionString matches with what is stored for the given version
+	if modelVersion.SoftwareVersionString != msg.SoftwareVersionString {
+		return nil, types.NewErrModelVersionStringDoesNotMatch(msg.Vid, msg.Pid, msg.SoftwareVersion, msg.SoftwareVersionString)
+	}
+
```
Comment (ashcherbakov) [inline x/compliance/keeper/msg_server_revoke_model.go:42]: I believe we need to fix the comments on lines 46 and 92 to not mention that `Model Info is not required to be on the ledger`

File: x/compliance/handler_test.go [hunk 2]
```diff
@@ -662,75 +692,48 @@ func TestHandler_CertifyRevokedModel(t *testing.T) {
...
-		secondCertifyModelMsg := NewMsgCertifyModel(
-			vid, pid, softwareVersion, softwareVersionString, certificationType, accAddress)
-		_, err = setup.Handler(setup.Ctx, secondCertifyModelMsg)
-		require.Error(t, err)
-		require.True(t, types.ErrAlreadyCertified.Is(err))
+		// query revoked model
+		revokedModel, _ := queryRevokedModel(setup, vid, pid, softwareVersion, certificationType)
+		require.False(t, revokedModel.Value)
 	}
 }
```
Comment (ashcherbakov) [inline x/compliance/handler_test.go:739]: Do we have a test that checks that a non-existent model can not be revoked?

File: x/compliance/keeper/msg_server_certify_model.go [hunk 3]
```diff
@@ -30,6 +30,17 @@ func (k msgServer) CertifyModel(goCtx context.Context, msg *types.MsgCertifyMode
...
+	}
+
+	// check if softwareVersionString matches with what is stored for the given version
+	if modelVersion.SoftwareVersionString != msg.SoftwareVersionString {
+		return nil, types.NewErrModelVersionStringDoesNotMatch(msg.Vid, msg.Pid, msg.SoftwareVersion, msg.SoftwareVersionString)
+	}
+
 	complianceInfo, found := k.GetComplianceInfo(ctx, msg.Vid, msg.Pid, msg.SoftwareVersion, msg.CertificationType)
 	// nolint:nestif
 	if found {
```
Comment (ashcherbakov) [inline x/compliance/keeper/msg_server_certify_model.go:46]: Please fix the comment on line 48 `The corresponding Model Info is not required to be on the ledger`

File: x/compliance/handler_test.go (hunk 2 above)
Comment (ashcherbakov) [inline x/compliance/handler_test.go:739]: Please add a couple of tests for Provisioning logic

**Review by ashcherbakov — APPROVED** [review]

File: x/compliance/handler_test.go (hunk 2 above)
Comment (spivachuk) [reply x/compliance/handler_test.go:739]: The test has been added in scope of https://github.com/zigbee-alliance/distributed-compliance-ledger/pull/182

File: x/compliance/keeper/msg_server_certify_model.go (hunk 3 above)
Comment (spivachuk) [reply x/compliance/keeper/msg_server_certify_model.go:46]: Fixed in scope of https://github.com/zigbee-alliance/distributed-compliance-ledger/pull/182

File: x/compliance/handler_test.go (hunk 2 above)
Comment (spivachuk) [reply x/compliance/handler_test.go:739]: Added in scope of https://github.com/zigbee-alliance/distributed-compliance-ledger/pull/182

**Pull request merged**